packages/twenty-server/src/modules/messaging/message-cleaner/commands/messaging-reset-channel.command.ts

```diff
@@ -0,0 +1,199 @@
+})
+export class MessagingResetChannelCommand extends CommandRunner {
+  private readonly logger = new Logger(MessagingResetChannelCommand.name);
+
+  constructor(
+    private readonly twentyORMGlobalManager: TwentyORMGlobalManager,
+    private readonly messagingMessageCleanerService: MessagingMessageCleanerService,
+    @InjectCacheStorage(CacheStorageNamespace.ModuleMessaging)
+    private readonly cacheStorage: CacheStorageService,
+  ) {
+    super();
+  }
+
+  async run(
+    _passedParam: string[],
+    options: MessagingResetChannelCommandOptions,
+  ): Promise<void> {
+    const { workspaceId, messageChannelId } = options;
+
+    const messageChannelRepository =
+      await this.twentyORMGlobalManager.getRepositoryForWorkspace<MessageChannelWorkspaceEntity>(
+        workspaceId,
+        'messageChannel',
+      );
+
+    if (messageChannelId) {
+      const messageChannel = await messageChannelRepository.findOne({
+        where: { id: messageChannelId },
+      });
+
+      if (!messageChannel) {
+        this.logger.error(
+          `Message channel ${messageChannelId} not found in workspace ${workspaceId}`,
+        );
+
+        return;
+      }
+
+      await this.resetMessageChannel(workspaceId, messageChannel);
+    } else {
+      this.logger.log(
+        `No message channel ID provided, resetting all message channels in workspace ${workspaceId}`,
+      );
+
+      const messageChannels = await messageChannelRepository.find();
+
+      if (messageChannels.length === 0) {
+        this.logger.log(
+          `No message channels found in workspace ${workspaceId}`,
+        );
+
+        return;
+      }
+
+      this.logger.log(
+        `Found ${messageChannels.length} message channels to reset`,
+      );
+
+      for (const messageChannel of messageChannels) {
+        await this.resetMessageChannel(workspaceId, messageChannel);
+      }
+
+      this.logger.log(
+        `Successfully reset all ${messageChannels.length} message channels in workspace ${workspaceId}`,
+      );
+    }
+  }
+
+  private async resetMessageChannel(
+    workspaceId: string,
+    messageChannel: MessageChannelWorkspaceEntity,
+  ): Promise<void> {
+    const messageChannelId = messageChannel.id;
+
+    this.logger.log(
+      `Starting reset for message channel ${messageChannelId} in workspace ${workspaceId}`,
+    );
+
+    const messageChannelRepository =
+      await this.twentyORMGlobalManager.getRepositoryForWorkspace<MessageChannelWorkspaceEntity>(
+        workspaceId,
+        'messageChannel',
+      );
+
+    const messageChannelMessageAssociationRepository =
+      await this.twentyORMGlobalManager.getRepositoryForWorkspace<MessageChannelMessageAssociationWorkspaceEntity>(
+        workspaceId,
+        'messageChannelMessageAssociation',
+      );
+
+    const associations = await messageChannelMessageAssociationRepository.find({
```

Comment:
this whole cleaning logic should be part of a service and be re-usable in this command but also when we remove a message channel from the UI. 
We already have a MessagingConnectedAccountDeletionCleanupJob that's doing:  await this.messageCleanerService.cleanOrphanMessagesAndThreads(
      data.workspaceId,
    );
We should try to unify